.pipelines/templates/aks-setup.yaml

```diff
@@ -1,27 +1,51 @@
 steps:
-  - script: |
-      echo "##vso[task.setvariable variable=AZURE_CLUSTER_NAME]sscd-e2e-$(openssl rand -hex 6)"
-      echo ${AZURE_CLUSTER_NAME}
-    displayName: "[AKS] Set cluster name"
-    condition: and(succeeded(), eq(variables['CLUSTER_TYPE'], 'aks'))
+  - script: |	
+      echo "##vso[task.setvariable variable=AZURE_CLUSTER_NAME]sscd-e2e-$(openssl rand -hex 6)"	
+      echo ${AZURE_CLUSTER_NAME}	
+    displayName: "[AKS Engine] Set cluster name"	
+    condition: and(succeeded(), eq(variables['CLUSTER_TYPE'], 'aks-engine'))	
 
   - script: |
-      az group create -n ${AZURE_CLUSTER_NAME} -l ${AZURE_LOCATION}
+      az group create -n ${AZURE_CLUSTER_NAME} -l $(AZURE_LOCATION)
 
-      az aks create \
-        -g ${AZURE_CLUSTER_NAME} \
-        -n ${AZURE_CLUSTER_NAME} \
-        --subscription ${SUBSCRIPTION_ID} \
-        --node-count ${AGENT_COUNT} \
-        --kubernetes-version ${AKS_KUBERNETES_VERSION} \
-        --node-vm-size Standard_D2_v3 \
-        --location ${AZURE_LOCATION} \
-        --service-principal ${CLIENT_ID} \
-        --client-secret ${CLIENT_SECRET} \
-        --windows-admin-username azureuser \
-        --windows-admin-password ${WINDOWS_ADMIN_PASSWORD} \
-        --network-plugin azure \
-        --no-ssh-key
+      if [[ ${LARGE_CLUSTER:-} == "true" ]]; then
+        az aks create \
+          -g ${AZURE_CLUSTER_NAME} \
+          -n ${AZURE_CLUSTER_NAME} \
+          --node-count $(AGENT_COUNT) \
+          --nodepool-name nodepool1 \
+          --kubernetes-version $(AKS_CLUSTER_VERSION) \
```

Comment:
should `--kubernetes-version` be defaulted? it gets outdated over time

Reply:
If we don't then it would always run the latest default in AKS?